src/commands/registers.ts

```diff
@@ -222,10 +234,386 @@ export class SomeRegisterCommand extends EmacsCommand {
     });
 
     if (copiedRectTexts) {
-      this.textRegisters.set(registerKey, { type: "rectangle", rectTexts: copiedRectTexts });
+      this.registers.set(registerKey, { type: "rectangle", rectTexts: copiedRectTexts });
     }
 
     this.emacsController.exitMarkMode();
     makeSelectionsEmpty(textEditor);
   }
+
+  // point-to-register, C-x r SPC <r>
+  public async runPoint(textEditor: vscode.TextEditor, registerKey: string): Promise<void> {
+    // Save all cursor positions as new Position objects to ensure immutability
+    const positions = textEditor.selections.map(
+      (selection) => new vscode.Position(selection.active.line, selection.active.character),
+    );
+    // Store positions with explicit type annotation
+    if (positions.length === 0) {
+      MessageManager.showMessage("No position to store");
+      return;
+    }
+    // Ensure all positions are valid VSCode Position objects
+    const validPositions = positions.filter((pos): pos is vscode.Position => pos instanceof vscode.Position);
+    if (validPositions.length === 0) {
+      MessageManager.showMessage("Invalid positions");
+      return;
+    }
+    const registerData: PositionRegisterData = {
+      type: "positions",
+      positions: validPositions,
+    };
+    this.registers.set(registerKey, registerData);
+    // Keep selections empty at current positions
+    makeSelectionsEmpty(textEditor);
+    MessageManager.showMessage("Position saved in register");
+  }
+
+  // jump-to-register, C-x r j <r>
+  public async runJump(textEditor: vscode.TextEditor, registerKey: string): Promise<void> {
+    if (!this.registers.has(registerKey)) {
+      MessageManager.showMessage("Register does not contain a position");
+      return;
+    }
+
+    const data = this.registers.get(registerKey);
+    if (data?.type !== "positions") {
+      MessageManager.showMessage("Register does not contain a position");
```

Comment:
The correct message is "Register doesn't contain a buffer position or configuration"